test/repositories/TagLookupCacheTest.scala

```diff
@@ -0,0 +1,49 @@
+package repositories
+
+import org.scalatest.{BeforeAndAfterEach, FlatSpec, Matchers}
+import utils.TagTestUtils._
+
+class TagLookupCacheTest extends FlatSpec with BeforeAndAfterEach with Matchers {
+
+  private val cache = TagLookupCache
+
+  override def beforeEach(): Unit = cache.allTags.set(Nil)
+
+  behavior of "TagLookupCache.insertTag()"
+
+  it should "insert new tag to cache" in {
+
+    val testTag = createTestTag("test", 1, 0)
+    cache.insertTag(testTag)
+
+    cache.allTags.get.contains(testTag) shouldEqual true
```

Comment:
Do we also want to assert about the cache state before insertion?

Reply:
sure i can add that

Reply:
done